### PR TITLE
DCOS-40043: toggle component

### PR DESCRIPTION
**Pull request by weblancaster**
Add toggle component used to switch between two
elements (String or JSX).

Closes DCOS-40043

### CODE REVIEW

File: packages/toggle/tests/Toggle.test.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import { Toggle } from "../";
+
+expect.addSnapshotSerializer(createSerializer(emotion));
+
+const primary = () => <div>primary</div>;
+const secondary = () => <div>secondary</div>;
+
+describe("Toggle", () => {
+  it("renders correctly", () => {
```
Comment (DanielMSchmidt) [inline packages/toggle/tests/Toggle.test.tsx]: Can you write a title that explains what you are testing here? Correctly is pretty vague.

File: packages/toggle/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Toggle Name
+
+Toggle documentation
```
Comment (DanielMSchmidt) [inline packages/toggle/README.md]: 🔥The documentation is missing.

File: packages/toggle/components/toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  public render() {
+    const { primaryContent, secondaryContent } = this.props;
+    let content = primaryContent;
+
+    if (!this.state.isPrimary) {
+      content = secondaryContent;
+    }
+
+    return (
+      <div className={style} onClick={this.handleClick}>
```
Comment (mperrotti) [inline packages/toggle/components/toggle.tsx]: This isn't keyboard-navigable as-is. We should either do this as a `input[type="checkbox"]` in disguise, or add the correct ARIA roles and a `tabindex` to it, as well as keyboard handlers to toggle the state when Enter or Spacebar are pressed.

File: packages/toggle/style.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { css } from "emotion";
+
+export const style = css`
+  display: inline-block;
```
Comment (mperrotti) [inline packages/toggle/style.ts]: I think we should let component consumers decide what the `display` is. Let's _just_ make this component responsible for toggling itself on and off.

File: packages/toggle/README.md (hunk 2 above)
Comment (weblancaster) [reply packages/toggle/README.md]: @DanielMSchmidt We don't have documentation for this component, I will come up with something very simple but I would not block this PR because of the documentation.

File: packages/toggle/tests/Toggle.test.tsx (hunk 1 above)
Comment (weblancaster) [reply packages/toggle/tests/Toggle.test.tsx]: Yes

File: packages/toggle/components/toggle.tsx (hunk 3 above)
Comment (weblancaster) [reply packages/toggle/components/toggle.tsx]: Agree, I will add aria.

File: packages/toggle/style.ts (hunk 4 above)
Comment (weblancaster) [reply packages/toggle/style.ts]: Sounds good.

File: packages/clickable/style.ts [hunk 5]
```diff
@@ -1,10 +1,7 @@
 import { css } from "emotion";
-import { coreColors } from "../shared/styles/color";
 
-const { white } = coreColors();
 export const outline = css`
   &:focus {
-    box-shadow: 0 0 0 1px ${white}, 0 0 0 3px currentColor;
```
Comment (weblancaster) [inline packages/clickable/style.ts]: Not sure why we had this bold outline.

File: packages/toggle/stories/Toggle.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+const readme = require("../README.md");
+
+const primary = () => <div>primary component</div>;
+const secondary = () => <div>secondary component</div>;
+
+storiesOf("Toggle", module)
+  .addDecorator(withReadme([readme]))
+  .addWithInfo("string", () => (
+    <Toggle primaryContent="Hello" secondaryContent="Bye" />
```
[the remark this answers is not in the document]
Comment (weblancaster) [reply packages/toggle/stories/Toggle.stories.tsx]: @mperrotti I have addressed the ARIA/role issue which discards the second code suggestion.

I was advocating for us to abstract the logic and component entirely, meaning the developer only need to pass the data and UI Kit takes care of displaying it.

I'm open to leaving a bit more flexible using the render prop interface but I would rather see us not giving that condition flexibility in this component.

[the remark this answers is not in the document]
Comment (weblancaster) [reply packages/toggle/stories/Toggle.stories.tsx]: What would you suggest for the props to replace `primaryContent` and `secondaryContent`

**Comment by DanielMSchmidt** [comment]
@weblancaster Could you combine the commits? I think introducing a feature and fixing it afterwards is a bit strange 😇

File: packages/toggle/stories/Toggle.stories.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (mperrotti) [reply packages/toggle/stories/Toggle.stories.tsx]: If `primary-` and `secondary-` are a familiar naming convention for toggle content/values, we should keep what you have instead of making people learn something new

Comment (mperrotti) [inline packages/toggle/stories/Toggle.stories.tsx]: Pass a `tabIndex` to the story - it defaults to -1

Not part of this PR, but something to consider for the Clickable component:
`:focus { outline: 0; }` hides any visual indication that the element has focus. Where might I file a ticket to improve the keyboard navigation for that component?

[the remark this answers is not in the document]
Comment (weblancaster) [reply packages/toggle/stories/Toggle.stories.tsx]: What do you think about `on` `off`?

Comment (weblancaster) [reply packages/toggle/stories/Toggle.stories.tsx]: I will remove in this PR because it's pretty small overhead

[the remark this answers is not in the document]
Comment (mperrotti) [reply packages/toggle/stories/Toggle.stories.tsx]: What about `contentOn` and `contentOff`? I think adding `content-`makes it clearer that those props accept content to be rendered. `on` and `off` seem a little mysterious to me

[the remark this answers is not in the document]
Comment (weblancaster) [reply packages/toggle/stories/Toggle.stories.tsx]: Sounds good.

**Comment by mperrotti** [comment]
Sorry, one more small thing:
I think I prefer the name `ContentToggle` or `ToggleContent`. It better describes what this component does, and we're less likely to run into a conflict if we ever create a component that we want to name `Toggle`

File: packages/clickable/style.ts [hunk 7]
```diff
@@ -1,10 +1,7 @@
...
     outline: 0;
```
Comment (weblancaster) [inline packages/clickable/style.ts]: @mperrotti I decided to leave it here and decide later what to  do about the `outline` because of design wise we do not want to show but a11y I would yes

Comment (mperrotti) [reply packages/clickable/style.ts]: Sounds good. Have you spoken with the design team about how they want to handle this styling? Or should I?
